casper/src/main/scala/coop/rchain/casper/api/BlockApiImpl.scala

```diff
@@ -600,10 +601,22 @@ class BlockApiImpl[F[_]: Concurrent: RuntimeManager: BlockDagStorage: BlockStore
       par: Par,
       blockHash: String,
       usePreStateHash: Boolean
-  ): F[ApiErr[(Seq[Par], LightBlockInfo)]] =
-    for {
-      block     <- BlockStore[F].getUnsafe(blockHash.unsafeHexToByteString)
+  ): F[ApiErr[(Seq[Par], LightBlockInfo)]] = {
+    val hashIsInValid = blockHash.length != BlockHash.HexLength
+    val dataF = for {
+      _         <- new Exception(s"$blockHash is not a valid block hash.").raiseError.whenA(hashIsInValid)
+      blockOpt  <- BlockStore[F].get1(blockHash.unsafeHexToByteString)
```

Comment:
Conversion from hex can fail e.g. `"0011mm".unsafeHexToByteString`.

Also what with hex `0x112233`?

Reply:
Bigger question for me is how we ended up using string for binary in protobuf in the first place.

Reply:
Take a look how it's done here.
https://github.com/rchain/rchain/blob/fa6a47b248b6fae131d9071171c4e4b1c195b22e/casper/src/main/scala/coop/rchain/casper/api/BlockApiImpl.scala#L475-L490

Reply:
`Bigger question for me is how we ended up using string for binary in protobuf in the first place.` I'm not sure wdy mean. Are you talking about hash being a string here and not bytes?
https://github.com/rchain/rchain/blob/cdae0514e4b0f58b756bd57ee07d82e85f1e91e8/models/src/main/protobuf/DeployServiceCommon.proto#L53-L57